golang/pkg/dagent/utils/docker.go

```diff
@@ -305,16 +305,17 @@ func DeployImage(ctx context.Context,
 	envList := EnvMapToSlice(envMap)
 	mountList := buildMountList(cfg, dog, deployImageRequest)
 
-	matchedContainer, err := dockerHelper.GetContainerByName(ctx, dog, containerName, false)
+	matchedContainer, err := dockerHelper.GetContainerByName(ctx, containerName)
 	if err != nil {
 		dog.WriteContainerState("", fmt.Sprintf("Failed to find container: %s", containerName))
 		return err
 	}
+
 	if matchedContainer != nil {
 		dog.WriteContainerState(matchedContainer.State)
 	}
 
-	err = dockerHelper.DeleteContainerByName(ctx, dog, containerName, false)
+	err = dockerHelper.DeleteContainerByID(ctx, dog, matchedContainer.ID)
```

Comment:
`matchedContainer` can be NIL, which crashes dagent.
I think `DeleteContainerByID` should be moved inside the if above.